rpcs3/rpcs3qt/game_list_frame.cpp

```diff
@@ -1065,18 +1065,26 @@ void game_list_frame::ShowContextMenu(const QPoint &pos)
 		mb->deleteLater();
 		if (mb->exec() == QMessageBox::Yes)
 		{
-			if (mb->checkBox()->isChecked())
+			const bool remove_caches = mb->checkBox()->isChecked();
+			if (fs::remove_all(currGame.path))
 			{
-				RemoveShadersCache(cache_base_dir);
-				RemovePPUCache(cache_base_dir);
-				RemoveSPUCache(cache_base_dir);
-				RemoveCustomConfiguration(currGame.serial);
-				RemoveCustomPadConfiguration(currGame.serial);
+				if (remove_caches)
+				{
+					RemoveShadersCache(cache_base_dir);
+					RemovePPUCache(cache_base_dir);
+					RemoveSPUCache(cache_base_dir);
+					RemoveCustomConfiguration(currGame.serial);
+					RemoveCustomPadConfiguration(currGame.serial);
+				}
+				m_game_data.erase(std::remove(m_game_data.begin(), m_game_data.end(), gameinfo), m_game_data.end());
+				LOG_SUCCESS(GENERAL, "Removed %s %s in %s", currGame.category, currGame.name, currGame.path);
+				Refresh(true);
+			}
+			else
+			{
+				LOG_ERROR(GENERAL, "Failed to remove %s %s in %s", currGame.category, currGame.name, currGame.path);
```

Comment:
Consider logging fs::g_tls_error in this case.

Reply:
Indeed, we do it elsewhere already (I think when logging installation failure), so why not here too.

Reply:
Done.
`E Failed to remove HDD Game SCOTT PILGRIM VS. THE WORLD: THE GAME in H:/dev/emulators/rpcs3/bin/dev_hdd0/game/NPUB30162 (Access violation)`